# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLContext.h
# Commit: e1b0906509ef
# Full Hash: e1b0906509efa6433978b9f980ec639b7eafbc6d
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Fix non-webgl CI tests. r=handyman
#   
#   * Revert some partial webgl+oop+vr code.
#   * More missing FuncScope.
#   * Fix compile errors.
# ==============================================================================

diff -r 7e2a2b1b416f -r e1b0906509ef dom/canvas/WebGLContext.h
--- a/dom/canvas/WebGLContext.h	Wed Jan 08 22:19:23 2020 +0000
+++ b/dom/canvas/WebGLContext.h	Wed Jan 08 22:19:26 2020 +0000
@@ -436,10 +436,14 @@
     nsCString text;
     text.AppendPrintf("WebGL warning: %s: ", FuncName());
 
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wformat-security"
+#ifdef __clang__
+#  pragma clang diagnostic push
+#  pragma clang diagnostic ignored "-Wformat-security"
+#endif
     text.AppendPrintf(fmt, args...);
-#pragma clang diagnostic pop
+#ifdef __clang__
+#  pragma clang diagnostic pop
+#endif
 
     GenerateErrorImpl(err, text);
   }
@@ -612,7 +616,7 @@
   void PixelStorei(GLenum pname, GLint param);
   void PolygonOffset(GLfloat factor, GLfloat units);
 
-  already_AddRefed<layers::SharedSurfaceTextureClient> GetVRFrame();
+  RefPtr<layers::SharedSurfaceTextureClient> GetVRFrame();
   void ClearVRFrame();
   void EnsureVRReady();
 
@@ -1290,10 +1294,14 @@
     nsCString msg;
     msg.AppendPrintf("WebGL perf warning: %s: ", funcName);
 
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wformat-security"
+#ifdef __clang__
+#  pragma clang diagnostic push
+#  pragma clang diagnostic ignored "-Wformat-security"
+#endif
     msg.AppendPrintf(fmt, args...);
-#pragma clang diagnostic pop
+#ifdef __clang__
+#  pragma clang diagnostic pop
+#endif
 
     GenerateErrorImpl(0, msg);
 